zipkin-sampler/src/main/scala/com/twitter/zipkin/sampler/AdaptiveSampleRate.scala

```diff
@@ -129,10 +129,10 @@ class AdaptiveSampleRate(
 
   val calculator =
     { v: Int => Some(buffer.pushAndSnap(v)) } andThen
-      new StoreRateCheck[Seq[Int]](storeRate, stats.scope("storeRateCheck"), log) andThen
+      new TargetStoreRateCheck[Seq[Int]](targetStoreRate, stats.scope("targetStoreRateCheck"), log) andThen
```

Comment:
the smoking gun was local references ending up in the global storage rate calculator. It makes no sense to check if the local store rate is set, then later blindly access targetStoreRate. In chasing this through, I found the more critical bug.